File: backend/src/main/java/um_backend/models/Patient.java
package um_backend.models;

import lombok.With;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@With
@Document("patients")
public record Patient(
        @Id
        String id,
        String firstname,
        String lastname,
        String dateOfBirth,
        String gender,
        EmergencyContact emergencyContact,
        String nationality,
        String maritalStatus,
        String primaryLanguage,
        String occupation,
        String insuranceNr,
        ContactInformation contactInformation,
        String healthDataId
) {
}

File: frontend/src/models/patient/Patient.ts
import {ContactInformation, EmergencyContact} from "./ContactInformation.ts";

export type Patient = {
    id: string;
    firstname: string;
    lastname: string;
    dateOfBirth: string;
    gender: string;
    emergencyContact: EmergencyContact;
    nationality: string;
    maritalStatus: string;
    primaryLanguage: string;
    occupation: string;
    insuranceNr: string;
    contactInformation: ContactInformation;
    healthDataId: string;
}

export type PatientToAdd = {
    firstname: string;
    lastname: string;
    dateOfBirth: string;
    gender: string;
    emergencyContact: EmergencyContact;
    nationality: string;
    maritalStatus: string;
    primaryLanguage: string;
    occupation: string;
    insuranceNr: string;
    contactInformation: ContactInformation;
}
